lib/github_service/commands/run_test.rb

```diff
@@ -0,0 +1,302 @@
+    #
+    # The following command:
+    #
+    #   @miq-bot run-tests manageiq#5678
+    #
+    # will create a commit with the .travis.yml changes:
+    #
+    #   @@ -14,6 +14,6 @@ matrix:
+    #      fast_finish: true
+    #    env:
+    #      global:
+    #   -  - REPOS=
+    #   +  - REPOS=manageiq#5678
+    #      matrix:
+    #   -  - TEST_REPO=
+    #   +  - TEST_REPO=manageiq-ui-classic#1234
+    #
+    class RunTest < Base
+      # The user calling the command
+      attr_reader :issuer
+
+      # The org of the issue this command originated from
+      attr_reader :org
+
+      # The (extra) repo(s) being targeted to be included in the test run
+      attr_reader :repos
+
+      # The repo(s) that will have the test suite run
+      attr_reader :test_repos
+
+      # The *-cross_repo-tests rugged instance
+      attr_reader :rugged_repo
+
+      # The arguments for the `run-test` command being called
+      attr_reader :value
+
+      restrict_to :organization
+
+      def self.test_repo_url
+        Settings.github.run_tests_repo.url
+      end
+
+      def self.test_repo_name
+        Settings.github.run_tests_repo.name
+      end
+
+      def self.test_repo_clone_dir
+        @test_repo_clone_dir ||= begin
+                                   url_parts = test_repo_url.split("/")[-2, 2]
+                                   repo_org  = url_parts.first
+                                   repo_dir  = test_repo_name
+                                   File.join(::Repo::BASE_PATH, repo_org, repo_dir)
+                                 end
+      end
+
+      # Name of the bot
+      def self.bot_name
+        Settings.github_credentials.username
+      end
+
+      # The new branch name for this particular run of the command (uniq)
+      def branch_name
+        @branch_name ||= "#{self.class.bot_name}-run-tests-#{issue.number}-#{SecureRandom.uuid}"
+      end
+
+      def run_tests
+        clone_repo_if_necessary
+        create_cross_repo_test_branch
+        update_travis_yaml_content
+        commit_travis_yaml_changes
+        push_commit_to_remote
+        create_cross_repo_test_pull_request
+      end
+
+      private
+
+      def _execute(issuer:, value:)
+        @issuer = issuer
+        parse_value(value)
+        return unless valid?
+
+        run_tests
+      end
+
+      def parse_value(value)
+        @value = value
+
+        @test_repos, @repos = value.split(/\s+including\s+/)
+                                   .map { |repo_list| repo_list.split(",") }
+
+        # Add the identifier for the PR for this comment to @repos here
+        @repos ||= []
+        @repos  << "#{issue.repo_name}##{issue.number}"
+
+        @test_repos.map! { |repo_name| normalize_repo_name(repo_name) }
+        @repos.map!      { |repo_name| normalize_repo_name(repo_name) }
+      end
+
+      def normalize_repo_name(repo)
+        repo.include?("/") ? repo : "#{issue.organization_name}/#{repo}"
```

Comment:
We handle this internally in [`CrossRepo::Repository`](https://github.com/ManageIQ/manageiq-cross_repo/blob/master/lib/manageiq/cross_repo/repository.rb), can we just pass through the value that the user gives us?

Reply:
Probably?

To be fair, I was just copying what was done by @Fryguy here in this case:

https://github.com/Fryguy/miq_bot/commit/dc0cc52ca6bbbfeaf5061c6681f4e6c53b9e85ab#diff-b351efdefa95964e43ea7fcabca89843R26

That said, I at least want to make sure this works as is from `sidekiq` before changing this, since this is only a problem of being "too thorough" at the moment.

Reply:
Going to pass on working on this for this first pass, as it is being used in other ways to validate.